docs/guides/circuits.rst

```diff
@@ -0,0 +1,439 @@
+
+If you would like to follow along with this guide, please create a new python file and start it with the imports we'll need to ensure the rest of the code examples can be run:
+
+.. literalinclude:: /examples/circuits_guide.py
+  :language: python
+  :start-at: from orquestra.quantum.circuits import (
+  :end-at: N_QUBITS =
+
+Notice we also defined a variable for the number of qubits we'll be working with. This will make some list comprehensions easier later.
+
+.. _creating_circuits:
+
+Creating a circuit
+------------------
+
+For this QAOA problem, the first part of the circuit we need to create is the initial state preparation circuit. That can be done by putting a Hadamard gate on each qubit. In the end we want the initial state preparation circuit to look like this:
+
+.. code-block:: text
+  
+       ┌───┐
+  q_0: ┤ H ├
+       ├───┤
+  q_1: ┤ H ├
+       ├───┤
+  q_2: ┤ H ├
+       └───┘
+
+We do this in our example using a `list comprehension <https://www.w3schools.com/python/python_lists_comprehension.asp>`_:
+
+.. literalinclude:: /examples/circuits_guide.py
+  :language: python
+  :start-at: state_prep_circ = Circuit(
+  :end-at: state_prep_circ = Circuit(
+
+We could have also explicitly put a Hadamard gate on each qubit index too, as we will see in the section on :ref:`appending circuits <appending_circuits>`.
+
+.. _getting_unitary:
+
+Getting the unitary matrix of a circuit
+---------------------------------------
+
+Now let's inspect some aspects of the unitary of the circuit, just as a sanity check to see that everything makes sense. We can convert our circuit to a unitary matrix with the ``.to_unitary()`` method. In this case, we should see that the shape of the unitary is an 8x8 matrix (because we have a circuit that operates on 3 qubits) and the type should be a numpy ndarray.
+
+.. literalinclude:: /examples/circuits_guide.py
+  :language: python
+  :start-at: unitary = state_prep_circ.to_unitary()
+  :end-at: ic(type(unitary))
+
+.. include:: /tutorials/beginner_tutorial.rst
+    :start-after: icecream-note:
+    :end-before: This will output a text description of the circuit
+
+The output we get from those two ``icecream`` statements should be
+
+.. code-block:: text
+
+  ic| unitary.shape: (8, 8)
+  ic| type(unitary): <class 'numpy.ndarray'>
+
+Inverting a circuit
+-------------------
+
+In some contexts, you may want to invert a circuit. For instance, if instead of :ref:`building a bell state <creating_basic_circuits>`, you want to perform a bell measurement. Or perhaps your circuit requires some `uncomputation <https://quantum.country/search#building-blocks-quantum-search>`_ that you don't want to build manually. To get the inverse of a circuit, just call the ``.inverse()`` method on it:
+
+.. literalinclude:: /examples/circuits_guide.py
+  :language: python
+  :start-at: = state_prep_circ.inverse()
+  :end-at: ic(np.allclose(inverse_unitary, np.eye(8)))
```

Comment:
I would change the naming in this example.
`invert_state_prep_circ` -> `inverse_state_prep_circ`/`inverse_state_prep_circ`. "invert" is a verb, doesn't fit here.
`invert_test_circ` -> I would call it sth like `combined_circ`? 
`inverse_unitary` -> this one's confusing. It's not really `inverse_unitary`, it's unitary of combined non-inverse and inverse. So I'd call it `combined_unitary` ?

Reply:
Ah yeah all good points. This was an artifact from what we initially planned to do with this example so the naming is definitely funky. Will fix!

Reply:
fixed in af446a1749110a289630084291990551053306a0